wgpu/src/renderer/widget/scrollable.rs

```diff
@@ -16,15 +17,57 @@ fn scrollbar_bounds(bounds: Rectangle) -> Rectangle {
     }
 }
 
+fn scroller_bounds(
+    bounds: Rectangle,
+    content_bounds: Rectangle,
+    background_bounds: Rectangle,
+    offset: u32,
+) -> Rectangle {
+    let ratio = bounds.height / content_bounds.height;
+    let scrollbar_height = bounds.height * ratio;
+    let y_offset = offset as f32 * ratio;
+
+    Rectangle {
+        x: background_bounds.x + f32::from(SCROLLBAR_MARGIN),
+        y: background_bounds.y + y_offset,
+        width: background_bounds.width - f32::from(2 * SCROLLBAR_MARGIN),
+        height: scrollbar_height,
+    }
+}
+
 impl scrollable::Renderer for Renderer {
-    fn is_mouse_over_scrollbar(
+    fn scrollbar_bounds(
+        &self,
+        bounds: Rectangle,
+        content_bounds: Rectangle,
+        offset: u32,
+    ) -> (Rectangle, Rectangle) {
+        let background_bounds = background_bounds(bounds);
+        let scroller_bounds =
+            scroller_bounds(bounds, content_bounds, background_bounds, offset);
+
+        (background_bounds, scroller_bounds)
+    }
```

Comment:
It may be a good idea to split this method into two: `scrollbar_bounds` and `scroller_bounds`.

Maybe we could calculate the scroller bounds in `iced_native` from the `scrollbar_bounds`? It shoud simplify the trait a bunch.

Reply:
> Although, maybe we could calculate the scroller bounds in `iced_native` from the `scrollbar_bounds`? It shoud simplify the trait a bunch.

In order to calculate the scroller bounds in `iced_native` we would need to know the `SCROLLBAR_MARGIN` which is defined in the renderer.